certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -443,15 +470,29 @@ def _make_server_ssl(self, vhost):
         :type vhost: :class:`~certbot_nginx.obj.VirtualHost`
 
         """
+        ipv6info = self.ipv6_info(self.config.tls_sni_01_port)
+
         # If the vhost was implicitly listening on the default Nginx port,
         # have it continue to do so.
         if len(vhost.addrs) == 0:
             listen_block = [['\n    ', 'listen', ' ', self.DEFAULT_LISTEN_PORT]]
             self.parser.add_server_directives(vhost, listen_block, replace=False)
 
+        ipv6_block = ['']
+        if vhost.ipv6_enabled():
+            ipv6_block = ['\n    ',
+                            'listen',
+                            ' ',
+                            '[::]:{0} ssl'.format(self.config.tls_sni_01_port)]
+            if not ipv6info[1]:
+                # ipv6only=on is absent in global config
+                ipv6_block.append(' ')
+                ipv6_block.append('ipv6only=on')
+
         snakeoil_cert, snakeoil_key = self._get_snakeoil_paths()
 
         ssl_block = ([
+            ipv6_block,
             ['\n    ', 'listen', ' ', '{0} ssl'.format(self.config.tls_sni_01_port)],
```

Comment:
Actually, why add it as a separate directive at all, instead of just a single `listen [::]:443 ssl ipv6only=off` if appropriate and `listen 443 ssl` otherwise? Does that interact poorly with FreeBSD networking or something?

Reply:
Yeah, dual stack socket is a Linux only thing, unfortunately.